### PR TITLE
[docs] Update the build instructions for Windows

### PR DESCRIPTION
**Pull request by brendanzagaeski**
Context: https://gitter.im/xamarin/xamarin-android?at=5c4a25ba7a0f4d5b1900f7a4
Context: https://gitter.im/xamarin/xamarin-android?at=5c4a9d968ce4bb25b80795d3

Add environment setup instructions to make it easier for contributors to
make sure they have a Windows build environment that has the expected
build dependencies.  The instructions were a little confusing before
this because the [Build Dependencies][0] page doesn't have a dedicated
section about Windows.

Add a link to the [Using Your Build][1] page to make it easier for
contributors to find the next step about how to build a Xamarin.Android
application after they have built `Xamarin.Android.sln`.

Mention the (relatively new) `Xamarin.Android.Build.Tasks.sln` as a way
to work around output assembly file locking when working on the build
tasks on Windows.

[0]: https://github.com/xamarin/xamarin-android/blob/master/Documentation/building/dependencies.md
[1]: https://github.com/xamarin/xamarin-android/blob/master/Documentation/workflow/UsingYourBuild.md

### CODE REVIEW

File: Documentation/building/windows-instructions.md [hunk 1]
```diff
@@ -8,44 +8,81 @@ MSBuild version 15 or later is required.
 
 # Building Xamarin.Android
 
- 1. Install the [build dependencies](dependencies.md).
+ 1. Run the [Visual Studio Installer](https://visualstudio.microsoft.com/vs/).
```
Comment (brendanzagaeski) [inline Documentation/building/windows-instructions.md]: For this first draft, I put all the instructions for setting up the build dependencies directly into this document, but I'm not sure if this is a better approach than adding a Windows section to the [Build Dependencies](https://github.com/xamarin/xamarin-android/blob/master/Documentation/building/dependencies.md) page. So I'd be happy to change that.

Comment (brendanzagaeski) [reply Documentation/building/windows-instructions.md]: I also made the instructions fairly specific to the current dependencies and versions. I'm happy to update them again when the first non-preview version of Visual Studio 2019 comes out, for example.

File: Documentation/building/windows-instructions.md [hunk 2]
```diff
@@ -8,44 +8,81 @@ MSBuild version 15 or later is required.
...
+
+      * **.NET Framework 4 &ndash; 4.6 development tools**
+      * **.NET Framework 4.6.2 development tools**
+      * **.NET Framework 4.7 development tools**
+      * **.NET Core 2.0 development tools**
+
+    The following items are also recommended:
+
+      * **.NET Framework 4.7.1 development tools**
+      * **.NET Framework 4.7.2 development tools**
```
Comment (jonathanpeppers) [inline Documentation/building/windows-instructions.md]: Ugh, so mental note we should standardize all `$(TargetFrameworkVersion)` if we can. So we don't have to install all of these...

We need to get on 4.7.2 anyway, because of changes in MSBuild in VS 2019.

File: Documentation/building/windows-instructions.md [hunk 3]
```diff
@@ -8,44 +8,81 @@ MSBuild version 15 or later is required.
...
         git clone https://github.com/xamarin/xamarin-android.git
 
- 3. Navigate to the `xamarin-android` directory
+ 8. Navigate to the `xamarin-android` directory
 
- 4. (Optional) [Configure the build](configuration.md).
+ 9. (Optional) [Configure the build](configuration.md).
 
- 5. Prepare the project:
+10. In a [Developer Command Prompt][developer-prompt], prepare the project:
```
Comment (jonathanpeppers) [inline Documentation/building/windows-instructions.md]: So another option is to just put `MSBuild.exe` in `%PATH%`, which is what I do. I use powershell and only resort to the developer command prompt for something really specific.

Maybe I could add a quick explanation on how to do this in a future PR.

Comment (brendanzagaeski) [reply Documentation/building/windows-instructions.md]: Sounds good. The related trick I've been using is to start a Developer Command Prompt and then run `powershell` as the first command in it so that I can use PowerShell commands and ReadLine handling but inherit the environment from the Developer Command Prompt. I've occasionally taken advantage of that approach to switch between the Release and Preview MSBuild versions, for example. I also sometimes use `devenv /edit` to open files for editing, so it's handy to have that in the path too.

**Comment by jonpryor** [comment]
I think I mentioned this on our 1:1, but I think that we should now "hard split" the documentation.  Instead of `Documentation/building/dependencies.md`, we should instead have `Documentation/building/unix/dependencies.md` and `Documentation/building/windows/dependencies.md`, and ditto for the `instructions.md` files, i.e. move `Documentation/building/windows-instructions.md` to `Documentation/building/windows/instructions.md`.

This would hopefully prevent developers from reading "the" `dependencies.md` file and installing Mono when they're on Windows.

**Comment by brendanzagaeski** [comment]
Changes in the latest force-push to address review comments:

  * Update the PR to split the build instructions by OS.

  * Adjust the text about the `$(TargetFrameworkVersion)` to be a bit more flexible in case we forget to update the page right away when the default `$(AndroidFrameworkVersion)` changes to `v10.0`.

File: Documentation/building/windows/dependencies.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Build Dependencies for Windows
+
+Building Xamarin.Android requires:
+
+  * An existing installation of the Xamarin.Android SDK and the Android SDK
```
Comment (jonpryor) [inline Documentation/building/windows/dependencies.md]: Why is this required?

File: Documentation/building/windows/dependencies.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  * The .NET Framework 3.5 &ndash; 4.7 development tools
```
Comment (jonpryor) [inline Documentation/building/windows/dependencies.md]: Wouldn't it be '.NET Framework v4.7.2'?  I'm not sure about the appropriate terminology.

File: Documentation/building/windows/dependencies.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  * Git for Windows
+  * The Java Development Kit (JDK)
+
+The recommended steps to install these dependencies are:
+
+ 1. Run the [Visual Studio Installer](https://visualstudio.microsoft.com/vs/).
+
+ 2. Under the **Workloads** tab, ensure that the **Mobile development with
```
Comment (jonpryor) [inline Documentation/building/windows/dependencies.md]: Still not sure why this would be *required* to build xamarin-android, given that we *download and install our own Android SDK + NDK* as part of our normal build process.

It may be *desirable* for other reasons to do this, but I don't think it's *required*.

File: Documentation/building/windows/dependencies.md (hunk 4 above)
Comment (jonathanpeppers) [reply Documentation/building/windows/dependencies.md]: I think nuget restore would likely fail.

This change might help this: https://github.com/xamarin/xamarin-android/commit/46f02511f8c91b133fb09fdcad2fa8fe9f2cc730

But it is only on master.

Comment (jonpryor) [reply Documentation/building/windows/dependencies.md]: If this is still the case, please file a bug.  We should *not* require a pre-existing system install of Xamarin.Android in order to build xamarin-android.

Additionally, if it *does* work on master, that's *fine*, as these docs are going into master. :-)

File: Documentation/building/windows/dependencies.md (hunk 5 above)
Comment (jonpryor) [reply Documentation/building/windows/dependencies.md]: It is the ".NET Framework 3.5" part which confuses me; .NET 3.5 is *ancient*.

Again, I don't know offhand what the appropriate names/terminology should be -- though *presumably* we'd want/need the [.NET Framework 4.7 Developer Pack](https://www.microsoft.com/en-us/download/details.aspx?id=55168)? -- but the "3.5" part threw me for a loop, at least.

We should ensure that we're using the correct names, regardless, and ideally in a manner which doesn't start with "Install Visual Studio...".

Comment (brendanzagaeski) [reply Documentation/building/windows/dependencies.md]: ### About .NET Framework 3.5

Here are a few of the places .NET Framework 3.5 is currently still used:

* *xamarin-android/external/Java.Interop/external/cecil/Mono.Cecil.props*
* *xamarin-android/external/opentk/Source/Bind/Generator.Bind.csproj*
* *xamarin-android/external/opentk/Source/Converter/Generator.Convert.csproj*
* *xamarin-android/external/mono/external/cecil/Mono.Cecil.props*
* *xamarin-android/external/mono/external/linker/cecil/Mono.Cecil.props*

If I recall correctly, I think it was *xamarin-android/external/Java.Interop/external/cecil/Mono.Cecil.props* or *xamarin-android/external/mono/external/cecil/Mono.Cecil.props* that caused the first build error when the .NET Framework 3.5 development tools weren't installed, so I'm not sure how many of the other occurrences of the target framework would eventually be used by the build process.

### About command line installation steps to work without the Visual Studio Installer

If we want a set of instructions that will work outside of the Visual Studio Installer, I can prepare a set like that too. Then we'd have a set of instructions for folks who are using Visual Studio, and a set for folks who'd rather just install the command line tools. I think it's still useful to include the instructions for the Visual Studio Installer in that case.

Comment (pjcollins) [reply Documentation/building/windows/dependencies.md]: To jump in on the the .Net Framework 3.5 discussion, I run into an error when building `r8.csproj` on Windows if this is not installed.
```
"C:\Users\peter\source\pj\xamarin-android\src\r8\r8.csproj" (default target) (28:2) ->
(GetReferenceAssemblyPaths target) ->
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\bin\Microsoft.Common.CurrentVersion.targets(1179,5): error MSB3645: .NET Framework v3.5 Service Pack 1 was not found. In order to targe
t ".NETFramework,Version=v2.0", .NET Framework v3.5 Service Pack 1 or later must be installed. [C:\Users\peter\source\pj\xamarin-android\src\r8\r8.csproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\bin\Microsoft.Common.CurrentVersion.targets(1179,5): error MSB3644: The reference assemblies for framework ".NETFramework,Version=v2.0"
 were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Not
e that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend. [C:\Us
ers\peter\source\pj\xamarin-android\src\r8\r8.csproj]
```

I also see a related prompt when loading Xamarin.Android.sln into VS2017:
image

These particular issues are likely addressed by https://github.com/xamarin/xamarin-android/pull/2666 however.

File: Documentation/building/windows/dependencies.md (hunk 6 above)
Comment (dellis1972) [reply Documentation/building/windows/dependencies.md]: One reason we *might* want to keep the mobile development is to provide templates. 
Currently if we are using OSS X.A I guess you have to create the entire project yourself manually. @brendanzagaeski is that correct?
